doc/extensions/zephyr/application.py

```diff
@@ -117,9 +127,20 @@ def run(self):
         source_dir = '/'.join(['..' for i in range(num_slashes + 1)])
         mkdir = 'mkdir' if num_slashes == 0 else 'mkdir -p'
 
+        # Shell prompt
+        prompt = '$>'
```

Comment:
Rather than this "all" prompt, how about just generating two code blocks with text mentioning the host-os (if host-os ``all``, otherwise just generate the code block for the specified host-os without the preceding text)

Linux/macOS commands:

    $ blah blah blah

Windows commands:

    > blah blah blah

Reply:
The docutils DOM is kind of annoying, so here are some pointers just in case:

To make the above work, you'll have to return a list of four nodes (nodes.paragraph, nodes.literal_block, nodes.paragraph, nodes.literal_block). The nodes.paragraphs will have to contain nodes.Text containing the actual inline text.

Example: https://github.com/mbolivar/zephyr/commit/b40d9c81b0b87711ff2cf8d8b9ef020ea7673083

Reply:
Thanks for the snippet! Will use it as a starting point

Reply:
Actually for simplicity I'm going to keep it all in a literal block for now, it looks good enough:

```
# For Linux/macOS
cd $ZEPHYR_BASE/samples/hello_world
# For Windows
cd %ZEPHYR_BASE%/samples/hello_world

# Make a build directory, and use cmake to configure a Ninja-based build system:
mkdir build && cd build
cmake -GNinja -DBOARD=arduino_due ..

# Now run ninja on the generated build system:
ninja
```